src/pybel/io/mongodb.py

```diff
@@ -0,0 +1,203 @@
+    # The mongo .find() method requires that sub-elements be notated 'parent.child'
+    concept_name = '.'.join((CONCEPT, NAME))
+    concept_identifier = '.'.join((CONCEPT, IDENTIFIER))
+    filter_ = {
+        TYPE: NODE,
+        concept_name: name,
+        concept_identifier: identifier,
+        FUNCTION: function,
+        VARIANTS: variants,
+    }
+    for key, value in list(filter_.items()):
+        if not value:
+            del filter_[key]
+
+    return list(collection.find(filter_))
+
+
+def _rm_mongo_keys(item: dict):
+    """Remove any dictionary keys used for internal MongoDB identification from the items.
+
+    :param item: A dictionary representing a node or an edge from a MongoDB collection
+    """
+    for prop in (ID, TYPE):
+        if prop in item.keys():
+            del item[prop]
+
+
+def get_edges_from_node(collection: Collection, node: Mapping[str, Any]) -> List[Mapping[str, Any]]:
+    """Return all the edges for the given node.
+
+    :param collection: A MongoDB collection within a database where a PyBEL graph has been stored
+    :param node: The node whose edges should be returned
+    :return: A list of the edges pointing to or from the given node
+    """
+    if not is_valid_node(node):
+        raise ValueError("Invalid node ", node)
+    # Remove the _id and type properties from the node (since they won't be included in the edge source/target information)
+    n = deepcopy(node)
+    _rm_mongo_keys(n)
+    # Find all the links where either the source or the target is node n
+    filter_ = {TYPE: LINK, '$or': [{SOURCE: n}, {TARGET: n}]}
+    return list(collection.find(filter_))
+
+
+def get_edges_from_criteria(
+    collection: Collection,
+    node_name: str = None,
+    node_identifier: str = None,
+    node_function: str = None,
+    node_variants: List[Variant] = None,
+) -> List[Tuple[Mapping[str, Any], List[Mapping[str, Any]]]]:
+    """Get all the edges for nodes that match the given criteria and return in a list of tuples.
+
+    :param collection: A MongoDB collection within a database where a PyBEL graph has been stored
+    :param name: The name of the desired node
+    :param identifier: The identifier of the desired node
+    :param function: The type of the desired node ("protein", "complex", etc)
+    :param variants: A list of variants that the desired node should contain.
+     Note: nodes that contain the variants in addition to specified variants will be matched.
+    :return: A list of tuples. The first element of each tuple is the node, and the second element is a list of the edges.
+    """
+    if not (node_name or node_identifier):
+        raise ValueError("Either a name or an identifier is required to get edges.")
+    matching_nodes = find_nodes(
+        collection,
+        name=node_name,
+        identifier=node_identifier,
+        function=node_function,
+        variants=node_variants
+    )
+    edges = []
```

Comment:
would a dictionary data structure work better? Nodes are hashable, so dict.items() would get you effectively the same thing as this too